src/components/Search/SearchPageHeader/SearchPageHeaderInput.tsx

```diff
@@ -144,6 +146,25 @@ function SearchPageHeaderInput({queryJSON, searchRouterListVisible, hideSearchRo
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
+    const prevQueryRef = useRef('');
+
+    const handleSearchAction = useCallback(
+        (value: string) => {
+            const queryWithoutFilters = getQueryWithoutFilters(value);
+            const prevQueryWithoutFilters = getQueryWithoutFilters(prevQueryRef.current);
+
+            // Only call handleSearch if the query (without filters) has changed or if there's no previous query
+            if (!handleSearch || (prevQueryRef.current && queryWithoutFilters === prevQueryWithoutFilters)) {
+                return;
+            }
```

Comment:
This logic is already in `SearchAutocompleteList`. It shouldn't be required here.